tests/io_determinant_hdf5.c

```diff
@@ -6,7 +6,7 @@
 
 #define TEST_BACKEND  TREXIO_HDF5
 #define TREXIO_FILE   "test_determinant.h5"
-#define RM_COMMAND    "rm -f " TREXIO_FILE
+#define RM_COMMAND    "rm -rf " TREXIO_FILE
```

Comment:
`rm -rf ` inside C code might be a bit dangerous because some people will execute the code as root. If we introduce a bug, like changeing the TREXIO_FILE into "/" with a sed, it can erase lots important data.
Maybe we can do something like:
```
#define RM_COMMAND    "mkdir .Trash && mv -- " TREXIO_FILE " .Trash/ && rm -rf -- .Trash/" TREXIO_FILE " && rm -d -- .Trash"
```

Reply:
1. Why would someone execute tests as a root? 
2. Why would we change the names in the existing and functioning tests?

I just checked your solution but it does not protect from removing the sensitive files, since you just move the target folder under `.Trash` and remove it there with the same `-rf` key. So if you take e.g. TREXIO_FILE "/usr" then you lose an entire directory with all subdirectories when you `mv` it.

Better solution is to first remove all `*.txt*` files non-recursively and then remove the empty directory with `rm -d`.

Reply:
1. By mistake: people do 'sudo make install', and they could make `sudo make install check`
2. You could imagine that a user for unknown reasons uses a script to make a `sed` in the generated files, and replaces TREXIO_FILE by `$HOME/ test_determinant.h5` (with a space inside). It could be because the user has not sufficient write permissions for the tests on a particular machine. With this simple mistake, we can erase the home directory....

Oh yes! You are right! my solution is not OK. Yours is much better.

Reply:
OK,I pushed the necessary changes to this PR.